THE JOURNAL OF BIOLOGICAL CHEMISTRY
© 2001 by The American Society for Biochemistry and Molecular Biology, Inc.

Vol. 276, No. 48, Issue of November 30, pp. 45193–45200, 2001
Printed in U.S.A.

14-3-3 Proteins Mediate an Essential Anti-apoptotic Signal*
Received for publication, June 27, 2001, and in revised form, September 17, 2001
Published, JBC Papers in Press, September 27, 2001, DOI 10.1074/jbc.M105971200

Shane C. Masters‡§ and Haian Fu§¶
From the ‡Graduate Program in Molecular and Systems Pharmacology and the §Department of Pharmacology,
Emory University School of Medicine, Atlanta, Georgia 30322

The 14-3-3 proteins are a family of highly conserved
eukaryotic regulatory molecules that play important
roles in many biological processes including cell cycle
control and regulation of cell death. They are able to
carry out these effects through binding and modulating
the activity of a host of signaling proteins. The ability of
14-3-3 to inhibit Bad and other proapoptotic proteins
argues that 14-3-3 can support cell survival. To examine
this issue in a global sense, a specific inhibitor of 14-33/ligand interactions, difopein, was used. Difopein expression led to induction of apoptosis. Studies using
various components of survival and death signaling
pathways were consistent with a vital role for 14-3-3/
ligand interactions in signal transduction from upstream pro-survival kinases to the core apoptotic machinery. Because these kinases often become activated
during oncogenesis, the effect of difopein on cell death
induced by antineoplastic drugs was examined. It was
found that difopein enhances the ability of cisplatin to
kill cells. These data support the model that 14-3-3,
through binding to Bad and other ligands, is critical for
cell survival signaling. Inhibition of 14-3-3 may represent a useful therapeutic target for treatment of cancer
and other diseases involving inappropriate cell
survival.

The 14-3-3 proteins are a family of conserved dimeric eukaryotic regulatory molecules. 14-3-3 proteins are known for
their ability to bind multiple cellular protein ligands, with
more than 70 binding partners identified to date. 14-3-3 binding proteins are very diverse and include kinases, phosphatases, receptors, structural proteins, and transcription factors
(see Ref. 1 for a review). Most 14-3-3 ligands require phosphorylation before 14-3-3 can interact with them, and consensus
phosphoserine containing 14-3-3 binding motifs have been defined (2, 3). The large number and diversity of 14-3-3 ligands
has led to the suggestion that 14-3-3 is involved in many
different cellular processes, including mitogenesis, cell cycle
control, and apoptosis.
Although 14-3-3 has some targets, such as Bad (4), that are
proximal components of the apoptotic machinery, the number
of 14-3-3 ligands that feed into apoptosis signaling pathways in
a more indirect manner is staggering. 14-3-3 binding proteins
* This work was supported by National Institutes of Health Grants
GM53165 and GM60033 (to H. F.) and by a Predoctoral Fellowship in
Pharmacology/Toxicology from the Pharmaceutical Research and
Manufacturers of America Foundation (to S. C. M.). The costs of publication of this article were defrayed in part by the payment of page
charges. This article must therefore be hereby marked “advertisement”
in accordance with 18 U.S.C. Section 1734 solely to indicate this fact.
¶ To whom correspondence should be addressed: Dept. of Pharmacology, Emory University School of Medicine, 1510 Clifton Rd. NE,
Atlanta, GA 30322. Tel.: 404-727-0368; Fax: 404-727-0365; E-mail:
hfu@emory.edu.
This paper is available on line at http://www.jbc.org

This is an Open Access article under the CC BY license.

such as Raf-1 (5), protein kinase C (6), phosphatidylinositol
3-kinase (7), and mitogen-activated protein kinase kinase kinase 1 (8) have been shown to have significant involvement in
signaling cell survival and death. Because of this, determination of the role of 14-3-3 in survival signaling demands a global
approach to controlling 14-3-3/ligand interactions. The first
steps in this direction have been taken, using 14-3-3 ligand
binding-defective mutants. Such mutants can have a dominant
negative effect on some 14-3-3-mediated processes (see Ref. 9
for an example), possibly by forming inactive heterodimers
with endogenous wild type 14-3-3 proteins. It was found that
mutant 14-3-3 proteins could sensitize cells to apoptosis induced by UV irradiation or serum withdrawal and that this
effect was mediated by p38 mitogen-activated protein kinase
(10). Such a result may be explained by the ability of 14-3-3 to
inhibit the death-inducing mitogen-activated protein kinase
kinase kinase homolog ASK1 (9), which is upstream of p38.
Although this work provides valuable insights into the function
of 14-3-3 in p38-regulated apoptosis, technical limitations related to the use of stable cell lines place restrictions on its
applicability to other 14-3-3-mediated processes.
The peptide R18 is a 20-mer isolated from a phage display
screen for its ability to bind 14-3-3␶ (11). This peptide was
found to interact with 14-3-3 very specifically; however, it does
not possess selectivity among the 14-3-3 isoforms. Notably, the
R18/14-3-3 interaction does not require phosphorylation of
R18, and R18 was one of the first documented 14-3-3 ligands to
possess this trait. However, this novelty also raised the possibility that R18 bound 14-3-3 in a manner unlike natural ligands. This issue was resolved when the co-crystal structures
of 14-3-3␨ with R18, a Raf-1-derived phosphopeptide (pS-Raf259; Ref. 12), and a phosphopeptide taken from the 14-3-3
binding epitope of polyoma virus middle T antigen (3) were
determined. The phosphate groups of pS-Raf-259 and middle T
antigen were found to contact a cluster of residues, including
Lys49, Arg56, Tyr128, and Arg127, in the charged side of the
conserved amphipathic ligand binding groove of 14-3-3. A core
motif in R18, 10WLDLE14, was found in a position similar to
that of the phosphopeptides, with negatively charged Asp12 and
Glu14 making contacts similar to those of phosphoserine. The
hydrophobic residues in the R18 core make extensive contacts
with the hydrophobic side of the amphipathic groove. Thus,
although R18 lacks phosphoserine, it interacts with 14-3-3 in a
manner very similar to phosphorylated ligands.
Because R18 shares a common binding site on 14-3-3 with
other ligands, it is expected that R18 could competitively interfere with 14-3-3/ligand interactions. Indeed, this has been
shown for several 14-3-3 target proteins in vitro, including
Raf-1 (11), ASK1 (9), and exoenzyme S (13). This phenomenon
is not restricted to R18, because phosphopeptides derived from
14-3-3 ligands have been similarly used on several occasions
(see Ref. 14 for an example). However, because R18 does not
require phosphorylation, we hypothesized that it could be ex-

45193

45194

14-3-3 Proteins Are Required for Cell Survival

pressed in cells from a DNA construct to inhibit 14-3-3/ligand
interactions without activation by a kinase and without sensitivity to phosphatases. Expression of R18 should globally inhibit 14-3-3 in an isoform-independent manner, making it useful for determination of the cellular roles of 14-3-3/ligand
interactions. By using 14-3-3 inhibitor peptides, we have found
that 14-3-3/ligand interactions are critical mediators of an antiapoptotic signal in multiple cell types. This signal is in part
delivered through maintenance of mitochondrial integrity. Disruption of 14-3-3 survival signaling may provide new opportunities for therapeutic intervention in diseases involving insufficient apoptosis.
EXPERIMENTAL PROCEDURES

Plasmids—The insert for pSCM110, coding for N-terminally Myctagged R18, was made by polymerase chain reaction against R18pGEX-2T (11) using the primers 5⬘-A AAG CTT ATG GAG CAG AAG
CTG ATC AGC GAG GAG GAC CTG GGA TCC GAA TTC GTC GAC
TCG GCC GAC GGG-3⬘ and 5⬘-A TCT AGA TCA CTC GAG CCC AGC
GGC CCC-3⬘ and subcloned into the vector pCR3.1-Uni using the Unidirectional Eukaryotic TA cloning kit (Invitrogen). A similar construct,
pSCM121, containing difopein in place of R18 was made by subcloning
the ⬃100-base pair SalI/XbaI fragment of pSCM110 into the ⬃5-kilobase pair XhoI/XbaI fragment of the same vector. The predicted protein
sequence of difopein without tag is SADGA (PHCVP RDLSW LDLEA
NMCLP) GAAGL DSADG A(PHCV PRDLS WLDLE ANMCL P)GAAG
LE, with the individual R18 monomers in parentheses. R18 and difopein were fused to an N-terminal enhanced yellow fluorescent protein
(EYFP)1 tag by subcloning the BamHI/ApaI fragments of pSCM110 and
pSCM121, respectively, into pEYFP-C1 (CLONTECH) cut with BglII/
ApaI to create pSCM136 and pSCM138. pSCM174, containing the mutant R18(Lys) was made using the QuikChange site-directed mutagenesis kit (Stratagene) with pSCM136 as a template and 5⬘-GGG ATC
TGT CGT GGC TTA AGC TTA AGG CGA ATA TG-3⬘ and its reverse
complement as primers. pSCM182 is a mammalian expression vector
for enhanced cyan fluorescent protein (ECFP), created by inserting the
NdeI/SmaI fragment from pECFP-C1 (CLONTECH) into NdeI/EcoRV
cut pcDNA3 (Invitrogen). This vector was adapted for use with the
Gateway system (Invitrogen) by insertion of the RfC cassette after
digestion with EcoRI and treatment with Klenow fragment, creating
pSCM184. DNA coding for the C-terminal 54 amino acids of exoenzyme
S (C54) was obtained by polymerase chain reaction against pLXZ200
using the primers 5⬘-G GGG ACA AGT TTG TAC AAA AAA GCA GGC
TCC GAC GAT GAC GAT AAG ATG GGA GTG ACC CG-3⬘ and 5⬘-GGG
GAC CAC TTT GTA CAA GAA AGC TGG GTA TCA GGC CAG ATC-3⬘.
This was placed in the Gateway system vector pDONR201 (Invitrogen)
using BP clonase according to the manufacturer’s protocol to create
pSCM186. A vector coding for C54 with an N-terminal ECFP tag
(pSCM187) was made using the Gateway LR reaction with pSCM186
and pSCM184. The predicted protein sequence for C54 is GVTRR
VLEEA ALGEQ SGHSQ GLLDA LDLAS KPERS GEVQE QDVRL
RMRGL DLA. Mammalian expression plasmids for human Bcl-2 and
Bcl-XL were generous gifts from Dr. Hong-Gang Wang.
Cell Culture—HEK 293, HeLa, and COS-7 cells were grown in Dulbecco’s modified Eagle’s medium (Mediatech) containing 10% fetal calf
serum (Atlanta Biologicals). A549 and DU145 cells were grown in RPMI
1640 (Mediatech) containing 10% fetal calf serum. All cell lines were
obtained from the American Type Culture Collection. Transfection was
accomplished using FuGENE 6 cationic lipid reagent (Roche Molecular
Biochemicals).
Immunoprecipitation—48 h post transfection, 8 ⫻ 105 HEK 293 cells
were resuspended in lysis buffer (150 mM NaCl, 10 mM HEPES, pH 7.5,
0.2% Nonidet P-40, 5 mM NaF, 5 mM Na4P2O7, 2 mM Na3VO4, 10
mg/liter aprotinin, 10 mg/liter leupeptin, 1 mM phenylmethylsulfonyl
fluoride), incubated on ice for 1 h, and centrifuged to remove insoluble
materials. The resulting lysates were precleared with 50 ␮l of 50%
protein G-Sepharose 4FF (Amersham Pharmacia Biotech). 1 ␮l of 4
g/liter mouse anti-FLAG M2 antibody (Sigma-Aldrich) was added to the
cleared lysates, and 1.5 h later 50 ␮l of 50% protein G-Sepharose 4FF
was added. This mixture was rocked gently for 1.5 h before the beads

1
The abbreviations used are: EYFP, enhanced yellow fluorescent
protein; ECFP, enhanced cyan fluorescent protein; ␤-gal, ␤-galactosidase; EGFP, enhanced green fluorescent protein; PBS, phosphate-buffered saline.

were washed three times with ice-cold lysis buffer and once with 50 mM
HEPES, pH 7.5. Bound proteins were eluted by boiling and detected by
Western blot.
Attachment-based Viability Assay—Cells seeded at 5 ⫻ 104/well were
grown in 24-well plates and transfected with a lacZ marker and various
test plasmids, using at least three wells for each condition. Forty-five
hours later the samples were washed twice with PBS to remove floating
and loosely attached cells and then lysed with 200 ␮l/well of Z buffer
(100 mM Na2HPO4, pH 6.95, 10 mM KCl, 1 mM MgSO4, 50 mM ␤-mercaptoethanol, 0.2% Nonidet P-40). After addition of 50 ␮l of 1 g/liter
chlorophenolred-␤-D-galactopyranoside (Roche Molecular Biochemicals)
in Z buffer, the solutions were transferred to a microtiter plate, and
A550 nm was determined using a kinetic plate reader (Molecular Devices). ␤-Galactosidase activity was determined as the slope of the best fit
line to the observed A550 nm versus time curve. This corresponds to the
viability of the transfected cells.
Morphology-based Assay for Cell Death—COS-7 cells were seeded at
3 ⫻ 105 in 35-mm dishes and transfected as for the attachment-based
assay. Twenty-four hours after transfection, the cells were fixed with
0.5% glutaraldehyde ⫹ 2% formaldehyde and stained for ␤-galactosidase using 5-bromo-4-chloro-3-indolyl-␤-D-galactopyranoside (X-gal)
(15). At least 500 stained cells in each sample were counted in a blind
fashion, with dead cells distinguished by their rounded, blebbed
appearance.
Caspase 3 Activity Assay—The caspase 3 assay kit, Colorimetric from
Sigma-Aldrich, which contains the substrate acetyl-DEVD-p-nitroaniline, was used according to the manufacturer’s protocol.
Flow Cytometric Measurement of DNA Content—Cells were seeded in
35-mm dishes at a density of 3 ⫻ 105 1 day prior to transfection with
various test DNAs in combination with pEGFP-F (20% of total DNA;
CLONTECH). This marker vector codes for an enhanced green fluorescent protein containing a farnesylation signal sequence at the C terminus. Twenty-four hours after transfection, the cells were trypsinized
and washed with PBS (130 mM NaCl, 20 mM phosphate, pH 7.5).
Samples of the washed cells were taken for analysis of protein expression by Western blot. The remaining cells were fixed by adding 3 ml of
ice-cold 95% ethanol and were stored at 4 °C overnight to allow small
fragments of DNA to diffuse out of the cells (16). The next day cells were
washed twice with PBS containing 1% bovine serum albumin (SigmaAldrich) and then resuspended in 0.8 ml of PBS, 1% bovine serum
albumin. To the cells was added 100 ␮l of 10 g/liter RNase A (SigmaAldrich) in PBS, 1% bovine serum albumin, and 100 ␮l of a DNA
binding dye (Sigma-Aldrich): either 0.1 g/liter 7-aminoactinomycin D or
0.5 g/liter propidium iodide. Both of these dyes behaved equivalently
and were used interchangeably. The samples were then stored in the
dark at room temperature for at least 30 min prior to running them on
a FACSort flow cytometer (Becton Dickinson). 1 to 3 ⫻ 105 events were
counted for each sample. The data were analyzed using WinMDI v2.8
(J. Trotter; Scripps Research Institute) to exclude debris, clumps, and
nontransfected cells.
RESULTS

Difopein Binds 14-3-3 in Cells and Inhibits 14-3-3/Ligand
Interactions—Before starting our study of the effect of 14-3-3/
ligand interaction disruption on survival, we first wanted to
develop a high affinity 14-3-3 antagonist. It was shown that
dimerization of a 14-3-3 binding phosphopeptide led to increased affinity, presumably through bidentate interaction
with dimeric 14-3-3 (3). Thus, we placed two R18 coding sequences separated by a sequence coding for a short peptide
linker in a mammalian expression vector to create difopein
(dimeric fourteen-three-three peptide inhibitor; Fig. 1). Difopein was readily expressed in HEK293 cells as an EYFP fusion
(Fig. 2). The ability of difopein to bind 14-3-3 in cells was
determined using an immunoprecipitation assay. 14-3-3 precipitates contained EYFP-difopein but not EYFP (Fig. 2). It was
not necessary to specially treat the cells to induce difopein
binding, as might be expected if phosphorylation-dependent
14-3-3 ligands were used.
The 14-3-3/Raf-1 interaction was used as a model system to
determine whether expressed difopein is capable of disrupting
14-3-3/ligand binding. Immunoprecipitated 14-3-3 was examined for the presence of endogenous Raf-1 (Fig. 2). In the
presence of EYFP, a significant amount of Raf-1 was found in

14-3-3 Proteins Are Required for Cell Survival

45195

FIG. 1. Domain structure of difopein. A, difopein contains two direct repeats of the R18 peptide separated by a
flexible linker. The sequence of difopein
without a tag is shown, with each R18
monomer shown in bold type and underlined. B, structural diagram of one proposed difopein binding mode. A 14-3-3
dimer, shown with monomers in light and
dark gray, contains two ligand-binding
grooves. It is possible that both R18
monomers of difopein can bind 14-3-3 simultaneously, creating a high affinity
complex. When either of the 14-3-3-binding grooves is occupied by an R18 monomer, that groove cannot bind endogenous
ligands. This figure is adapted from Fu et
al. (1).

FIG. 2. Difopein expressed in cells
can bind 14-3-3 and inhibit 14-3-3/ligand interactions. HEK293 cells were
transfected with DNA coding for FLAGtagged 14-3-3␨, EYFP, or EYFP-difopein
as indicated. 14-3-3 was immunoprecipitated using anti-FLAG (M2; Sigma-Aldrich). The presence of various species in
the lysates and precipitated complexes
was determined by Western blot using anti-Raf-1 (sc-133; Santa Cruz Biotechnology; top panels), anti-GFP (8367; CLONTECH; middle panels), and anti-14-3-3
(sc-629; Santa Cruz Biotechnology; bottom panel). The anti-GFP antibody recognizes GFP, EGFP, ECFP, and EYFP.

the 14-3-3 complex; however, expression of EYFP-difopein reduced Raf-1 to nearly undetectable levels. These data indicate
that difopein is an effective competitor of the 14-3-3/Raf-1 interaction in cells. Based on this result and on the finding that
the conserved amphipathic groove of 14-3-3 is the primary
binding site for R18 (12) as well as physiological ligands (17,
18), it is reasonable to propose that difopein can disrupt many,
if not all, of the 14-3-3/ligand interactions that occur in cells.
Difopein Expression Induces Apoptotic Cell Death—The ability of 14-3-3 to bind and inhibit several proapoptotic proteins
supports the idea that 14-3-3 acts to promote cell survival.
However, 14-3-3 also binds many proteins not known to be
involved in cell death, and the overall effect of 14-3-3 on survival is not known. We used the difopein 14-3-3 inhibitor in
several cell death assays to address this issue. For the first
method, COS-7 cells were co-transfected with Myc-difopein and
a ␤-galactosidase (␤-gal) marker. Later, floating, dead cells

were gently washed away, and the attached cells were assayed
for ␤-gal. Cell death is detected as a decrease in the ␤-gal
activity relative to the vector control. Transfection with Mycdifopein led to a decrease in ␤-gal activity and thus viability
(Fig. 3A). This supports the hypothesis that 14-3-3/ligand interactions are required for critical cellular processes, possibly
including pro-survival signaling. Similar results were seen in a
cell morphology based assay (Fig. 3B).
Because disruption of survival signaling is expected to induce apoptosis rather than necrosis, it is essential to characterize the nature of the cell death caused by R18. Caspase 3 is
a downstream effector protease that is commonly activated
during apoptosis. Upon expression of EYFP-difopein, but not
EYFP, the levels of activated caspase 3 increase in COS-7 cells
(Fig. 3C), supporting the hypothesis that difopein induces apoptosis. We also used a flow cytometry based DNA content assay
to determine the population of cells containing cleaved DNA.

45196

14-3-3 Proteins Are Required for Cell Survival

FIG. 3. Disruption of 14-3-3/ligand interactions by difopein induces apoptosis. A, cell attachment assay. COS-7 cells were co-transfected
with difopein or a control vector (80% of DNA) along with a lacZ reporter gene (20% of DNA). After 45 h the floating cells were gently washed away,
and the attached cells were lysed and assayed for ␤-galactosidase. Decreased ␤-gal activity indicates a loss of viability. The results are normalized
to the control vector ␤-gal activity (mean ⫾ S.E.; n ⫽ 3). B, cell morphology assay. COS-7 cells were transfected as in A. Twenty-four hours later,
the cells were fixed and stained for ␤-gal. Transfected (␤-gal-positive) cells in each sample were examined microscopically for a rounded, blebbed
morphology indicative of death. The ratios above the bars are live cells/total cells counted. C, caspase-3 activity assay. Twelve hours after
transfection, COS-7 cells were lysed and assayed for the ability to cleave acetyl-DEVD-p-nitroaniline using the caspase 3 assay kit, Colorimetric
(Sigma-Aldrich). The results shown represent the mean caspase activity ⫾ S.E. (n ⫽ 3). D, DNA content distribution. COS-7 cells were transfected
with plasmids coding for EYFP-difopein or EYFP along with a farnesylated EGFP marker. Twenty-four hours later, the cells were fixed in ethanol
overnight, stained with 7-aminoactinomycin D, and run on a flow cytometer. Histograms for the 7-aminoactinomycin D signal, representing DNA
content, are shown (left panel) for transfected cells (EGFP-F-positive) only. Histograms are normalized to the total number of transfected cells such
that the area under each curve is 100. The right panel shows the fraction of cells containing sub-G0 DNA content, which represent apoptotic cells
with fragmented DNA. The results shown are representative of at least three independent experiments.

Internucleosomal DNA cleavage, which is a commonly used
marker of apoptosis, produces small fragments that can diffuse
out of cells during ethanol fixation, resulting in decreased
signals being seen on staining for DNA (sub-G0 DNA content).
In COS-7 cells, EYFP-difopein caused a dramatic increase in
the fraction of sub-G0 cells (Fig. 3D), whereas EYFP behaved
similarly to empty vector (data not shown). Thus, difopeininduced disruption of 14-3-3/ligand interactions can cause apoptosis in COS-7 cells, supporting a role of 14-3-3 in survival
signaling.
It is possible that the effects of 14-3-3 inhibition on apoptosis
are restricted to COS-7 cells. Three additional cell lines were

used to examine this issue: A549 lung cancer cells, DU145
prostate cancer cells, and HeLa cervical carcinoma cells. In the
attachment-based viability assay, Myc-difopein was found to
kill all three of these cell lines to varying degrees (Fig. 4),
supporting the generality of the difopein effect. Because 14-3-3
proteins are ubiquitously expressed and because 14-3-3 can
potentially target many different apoptosis regulating molecules, we anticipate that most cells will show some degree of
sensitivity to 14-3-3 inhibitors.
Difopein Causes Cell Death through 14-3-3 Inhibition—Although the parent molecule of difopein, R18, is highly specific
for 14-3-3, binding no other proteins in radiolabeled cell lysates

14-3-3 Proteins Are Required for Cell Survival

FIG. 4. Difopein is active in multiple cell lines. A549, DU145,
and HeLa cells were transfected and assayed for ␤-galactosidase as in
Fig. 3A. The ␤-gal activity for each cell line is relative to the vector
control activity for that line and represents the mean ⫾ S.E. (n ⫽ 3).

(11), this type of approach is limited by the possibility that
critical proteins of low abundance will not be detected. We have
developed two lines of evidence that strongly support the hypothesis that the effects of difopein on cell death are caused by
inhibition of 14-3-3. First, we changed two acidic residues of
R18 (Asp12 and Glu14), which are involved in coordinating the
basic cluster of 14-3-3 (12), to lysines, creating R18(Lys). This
peptide is not expected to bind 14-3-3, but it could retain
interaction with other cellular partners if they exist. Indeed,
EYFP-R18(Lys) was unable to be co-precipitated with 14-3-3␨
from HEK293 cells even under conditions where EYFP-R18
could be (Fig. 5A). When tested for its ability to induce DNA
fragmentation, EYFP-R18(Lys) behaved essentially identically
to EYFP and did not induce apoptosis (Fig. 5B), despite its
robust expression (data not shown). The second approach used
a 14-3-3 binding molecule that is structurally unrelated to
difopein, which should therefore have a different spectrum of
non-14-3-3-specific protein interactions. We have characterized
the phosphorylation-independent interaction of 14-3-3 with
Pseudomonas aeruginosa exoenzyme S (13, 18, 19), finding that
it binds in the amphipathic groove of 14-3-3 with high affinity.
Others later determined that the 14-3-3 binding epitope of
exoenzyme S is localized to its C-terminal 54 residues (20). This
domain, termed C54, was able to bind 14-3-3 (Fig. 5A) and
induce apoptosis in COS-7 cells as determined by DNA content
(Fig. 5C). Thus, a strong correlation exists between the abilities
of these molecules to bind 14-3-3 and to cause cell death. This
argues that the apoptosis caused by difopein is due to inhibition
of 14-3-3/ligand interactions and that 14-3-3, by binding other
cellular proteins, mediates a critical prosurvival signal.
The 14-3-3 Survival Signal Acts Upstream of Mitochondria—
Because apoptosis is irreversible once a critical commitment step
is reached, the initiation of this process is subject to multiple
levels of regulation. To place the 14-3-3 survival signal among
other known death regulators, the effect of overexpression or
inhibition of several regulators on EYFP-difopein-induced apoptosis was examined. First, the pan-caspase inhibitor zVAD-fmk
was used to disable both upstream signaling caspases as well as
downstream effector caspases. Treatment of COS-7 cells with
zVAD-fmk at 50 ␮M abolished the ability of EYFP-difopein to kill
cells (Fig. 6A). This result is most likely explained by a requirement for effector caspases in the apoptosis caused by disruption

45197

of 14-3-3/ligand interactions and is consistent with the ability of
difopein to activate caspase 3 (Fig. 3C).
The Bcl-2 proteins are essential regulators of apoptosis induced by many different stimuli. This family, which contains
both pro- and anti-apoptotic members, acts in part by controlling the status of the mitochondria. Two antiapoptotic Bcl-2
homologs, Bcl-2 and Bcl-XL, were transfected into COS-7 cells
to determine their ability to block EYFP-difopein-induced
death. Both Bcl-XL (Fig. 6B) and Bcl-2 (data not shown) were
able to strongly inhibit difopein-induced apoptosis, even under
conditions where they had relatively small effects on death
caused by the proapoptotic Bcl-2 homolog Bad (data not
shown). These results suggest that the 14-3-3 survival signal
acts upstream of the Bcl-2 proteins, possibly aiding in the
maintenance of mitochondrial integrity.
Protein phosphorylation is a common mechanism for regulating cellular processes, including apoptosis, and many kinases are known to promote (i.e. ASK1; Ref. 21) or inhibit (i.e.
Akt/PKB; Refs. 22–24) cell death. Because phosphorylation of
serine or threonine is usually required to induce 14-3-3/ligand
interaction, it is possible that the effects of difopein on cellular
processes would be insensitive to kinase activity. This was
tested for a key survival signaling kinase, Akt. As predicted,
constitutively active Akt had little effect on EYFP-difopeininduced apoptosis (Fig. 6C), despite its robust expression (data
not shown) and its ability to decrease the basal level of cell
death. Together, the data presented in this section suggest that
14-3-3/ligand interactions are important for signal transmission from upstream survival signaling kinases to the mitochondria, although it is likely that other roles for 14-3-3 in regulating apoptosis await discovery.
Disruption of 14-3-3/Ligand Interactions Can Enhance Sensitivity to the Antineoplastic Agent Cisplatin—During oncogenesis, most cancers develop defects in apoptosis regulation (25,
26), often because of activation of upstream survival signaling
pathways. Because conventional antineoplastic agents rely on
induction of apoptosis for their efficacy (27), agents that can
blunt this overactivation of survival mechanisms may enhance
the utility of current anticancer therapies. Treatment of COS-7
cells with the antitumor drug cisplatin caused a dose-dependent decrease in viability as measured by an attachment-based
assay (Fig. 7 and data not shown). Transfection of the cells with
Myc-R18 prior to cisplatin treatment enhanced the ability of
cisplatin to kill cells (Fig. 7). It was possible to see a decrease in
viability caused by the combination of R18 and cisplatin at
doses where each alone produced no detectable effect, and at
higher treatment levels the combination was still more active
than either agent alone. In addition, it was possible to see
similar effects in the HeLa cervical cancer cell line (data not
shown). These results are encouraging and show that in principle inhibition of 14-3-3/ligand interactions could be a useful
therapeutic strategy for the treatment of cancer.
DISCUSSION

The 14-3-3 proteins are biochemically well characterized as
phosphoserine-binding proteins (2). However, the large number and diversity of 14-3-3 ligands has made 14-3-3 difficult to
classify in terms of function. The data presented here strongly
support a role for 14-3-3 in prevention of apoptosis through
transduction of survival signals. Expression of 14-3-3/ligand
interaction inhibitors in cells led to the induction of apoptosis.
These results indicate that 14-3-3 proteins are essential in
mammalian cells, consistent with results of 14-3-3 knockout in
yeast (28 –30). To some degree our findings contrast with a
previous study using ligand binding-defective mutants of 143-3 (10). These mutants have dominant negative effects in
some systems, presumably via heterodimerization with wild

45198

14-3-3 Proteins Are Required for Cell Survival

FIG. 5. The ability of peptides to
bind 14-3-3 is correlated with their
ability to induce apoptosis. A, various
peptides were assayed for FLAG-14-3-3␨
binding by immunoprecipitation (IP) as in
Fig. 2. The peptides in lanes 3–5 and 8
were expressed as EYFP fusions, whereas
those in lanes 7 and 9 were ECFP fusions.
The top and bottom panels were blotted
with anti-GFP (8367; CLONTECH)
whereas the middle panel was blotted
with anti-FLAG (M2; Sigma-Aldrich). B
and C, the ability of R18(Lys) (B) or C54
(C) to induce COS-7 cell death was determined as in Fig. 3D. Western blot (not
shown) was used to verify the expression
of all peptides.

type 14-3-3. It was reported that dominant negative 14-3-3 can
enhance the death caused by some stimuli, such as UV irradiation, but by itself cannot induce apoptosis (10). It is likely that
part of this difference is due to the use of stable cell lines. The
14-3-3 mutant proteins were continuously expressed during the
generation of these lines, so that only clones that developed
resistance to the proapoptotic effects of 14-3-3 inhibition could
be isolated. An additional difficulty is that the mutants used
may have heterogenous effects on ligand binding affinity with
respect to different ligands (17),2 making it possible that only a
small subset of 14-3-3/ligand interactions were disrupted.
Thus, the use of a global inhibitor of 14-3-3 in transient transfection systems is expected to provide a more complete view of
the role of 14-3-3 in apoptosis.
One of the most important issues related to the use of 14-33/ligand interaction inhibitors is that of their specificity. The
2

H. Wang and H. Fu, unpublished results.

R18 peptide is highly specific for binding to 14-3-3 proteins
(11). In addition, the abilities of R18 to bind 14-3-3 and induce
apoptosis were both abolished by mutation of two residues
shown to be critical for contacting 14-3-3 (12). This argues
strongly that the phenotypes seen upon R18 expression require
14-3-3 binding. These results also support the hypothesis that
difopein is acting through specific binding and inhibition of
14-3-3. However, it is impossible to completely rule out the
presence of 14-3-3-independent effects of R18 or difopein. Thus,
we tested the structurally unrelated 14-3-3 inhibitor C54 and
found that it could induce apoptosis. Because of its lack of
homology to difopein, it is unlikely that difopein and C54 share
common non-14-3-3-specific effects. Together our results support the interpretation that the phenotypes induced by difopein
reflect a loss of 14-3-3/ligand interactions.
A multitude of proteins have been described that positively
or negatively regulate the induction of apoptosis. 14-3-3 can be
placed in this network as acting upstream of the Bcl-2 proteins

14-3-3 Proteins Are Required for Cell Survival

45199

FIG. 6. Placement of the 14-3-3 survival signal in the apoptosis signaling network. A, COS-7 cells were
treated with the pan-caspase inhibitor
zVAD-fmk at 50 ␮M or with vehicle
(Me2SO) prior to transfection with EYFP
or EYFP-difopein. DNA fragmentation
was used to determine apoptosis as in Fig.
3D. B and C, the effects of Bcl-XL (B) and
constitutively active Akt (C) on difopeininduced COS-7 cell death were measured
as in Fig. 3D.

FIG. 7. Cisplatin and R18 decrease the viability of COS-7 cells
in an additive fashion. COS-7 cells were co-transfected with R18
DNA (amount used shown as the fraction of total DNA) and a lacZ
marker gene. 12 h after transfection, the indicated concentrations of
cisplatin were added. Approximately 36 h later the viability of the cells
(mean ⫾ S.E., n ⫽ 3) was determined as in Fig. 3A.

and effector caspases but downstream or independent of the
pro-survival kinase Akt. This is consistent with the finding
that Akt inhibition of the proapoptotic Bcl-2 homolog Bad re-

quires 14-3-3 (32). It is possible that difopein acts to disrupt the
14-3-3/Bad complex and liberate free, active Bad. Although this
simple model may account for a portion of the 14-3-3 effect on
cell survival, the large number of 14-3-3 ligands directly or
indirectly involved in cell death regulation makes it unlikely
that it can completely explain the ability of difopein to induce
apoptosis.
Studies of oncogenesis have shown that transforming mutations in preneoplastic cells often predispose them to apoptosis,
either intrinsic to the cell or mediated by tumor surveillance
systems in the host (27, 31). To compensate for this, many
cancers develop enhanced antiapoptotic survival signaling networks (26). This provides a compelling rationale to support the
targeting of apoptosis in general and survival signaling in
particular for the treatment of cancer. Additionally, many currently available antineoplastic therapies act via induction of
apoptosis (27), further supporting such a strategy. The ability
of difopein to inhibit the antiapoptotic activity of the survival
signaling kinase Akt and the additive effect of difopein on
cisplatin-induced death suggest that blockade of 14-3-3/ligand
interactions could be a therapeutically relevant target for cancer. It is interesting to note that the tumor suppressor protein
p53 is not required for difopein-induced cell death, because
difopein is active in COS-7 cells, which express the p53 inactivating SV40 T antigen, and in DU145 cells, which are mutated
in the p53 gene. This is an important point because many
antineoplastic agents are relatively ineffective against cells

45200

14-3-3 Proteins Are Required for Cell Survival

that lack p53 and because p53 mutation is extremely common
in cancer. Mechanistically, the additive effects of 14-3-3 inhibitors and cisplatin are consistent with the hypothesis that
disruption of the 14-3-3 survival signal sensitizes cells to cisplatin-induced apoptosis.
Although the work described here focused on apoptotic cell
death, 14-3-3 proteins are thought to be involved in many other
cellular processes. The ability of 14-3-3 inhibitors to induce
apoptosis does not imply that they produce only this phenotype.
For example, expression of difopein can blunt the accumulation
of HeLa cells in the S phase of the cell cycle caused by treatment with low concentrations of doxorubicin (data not shown).
It may be possible to use low levels of 14-3-3 inhibitors, cells
partially resistant to difopein-induced cell death, or treatment
with apoptosis inhibitors such as Bcl-2 to examine the role of
14-3-3/ligand interactions in modulating other actions of the
cell.
In summary, through the use of inhibitors of 14-3-3/ligand
interactions, it has been shown that 14-3-3 proteins are important mediators of survival signals. This suggests that the 143-3 proteins are essential in mammalian cells, as has been
previously shown in yeast. This work may have practical applications in treatment of diseases involving disregulated apoptosis and could also aid in the dissection of the function of
14-3-3 in other cellular processes.
Acknowledgments—We thank Drs. Bingcheng Wang and Hong-Gang
Wang for the generous gifts of DNA constructs and the members of the
Fu laboratory for helpful discussions.
REFERENCES
1. Fu, H., Subramanian, R. R., and Masters, S. C. (2000) Annu. Rev. Pharmacol.
Toxicol. 40, 617– 647
2. Muslin, A. J., Tanner, J. W., Allen, P. M., and Shaw, A. S. (1996) Cell 84,
889 – 897
3. Yaffe, M. B., Rittinger, K., Volinia, S., Caron, P. R., Aitken, A., Leffers, H.,
Gamblin, S. J., Smerdon, S. J., and Cantley, L. C. (1997) Cell 91, 9619 –9671
4. Zha, J., Harada, H., Yang, E., Jockel, J., and Korsmeyer, S. J. (1996) Cell 87,
619 – 628
5. Salomoni, P., Wasik, M. A., Riedel, R. F., Reiss, K., Choi, J. K., Skorski, T., and
Calabretta, B. (1998) J. Exp. Med. 187, 1995–2007
6. Reyland, M. E., Barzen, K. A., Anderson, S. M., Quissell, D. O., and Matassa,

A. A. (2000) Cell Death Differ. 7, 1200 –1209
7. Toker, A. (2000) Mol. Pharmacol. 57, 652– 658
8. Widmann, C., Gerwins, P., Johnson, N. L., Jarpe, M. B., and Johnson, G. L.
(1998) Mol. Cell. Biol. 18, 2416 –2429
9. Zhang, L., Chen, J., and Fu, H. (1999) Proc. Natl. Acad. Sci. U. S. A. 96,
8511– 8515
10. Xing, H., Zhang, S., Weinheimer, C., Kovacs, A., and Muslin, A. J. (2000)
EMBO J. 19, 349 –358
11. Wang, B., Yang, H., Liu, Y. C., Jelinek, T., Zhang, L., Ruoslahti, E., and Fu, H.
(1999) Biochemistry 38, 12499 –12504
12. Petosa, C., Masters, S. C., Bankston, L. A., Pohl, J., Wang, B., Fu, H., and
Liddington, R. C. (1998) J. Biol. Chem. 273, 16305–16310
13. Masters, S. C., Pederson, K. J., Zhang, L., Barbieri, J. T., and Fu, H. (1999)
Biochemistry 38, 5216 –5221
14. Thorson, J. A., Yu, L. W., Hsu, A. L., Shih, N. Y., Graves, P. R., Tanner, J. W.,
Allen, P. M., Piwnica-Worms, H., and Shaw, A. S. (1998) Mol. Cell. Biol. 18,
5229 –5238
15. Sanes, J. R., Rubenstein, J. L., and Nicolas, J. F. (1986) EMBO J. 5, 3133–3142
16. Amarante-Mendes, G. P., Bossy-Wetzel, E., Brunner, T., Finucane, D., Green,
D. R., and Kasibhatla, S. (1998) in Cells: A Laboratory Manual (Spector,
D. L., Goldman, R. D., and Leinwand, L. A., eds) Vol. 1, pp. 15.1–15.24, Cold
Spring Harbor Laboratory, Cold Spring Harbor, NY
17. Wang, H., Zhang, L., Liddington, R., and Fu, H. (1998) J. Biol. Chem. 273,
16297–16304
18. Zhang, L., Wang, H., Liu, D., Liddington, R., and Fu, H. (1997) J. Biol. Chem.
272, 13717–13724
19. Zhang, L., Wang, H., Masters, S. C., Wang, B., Barbieri, J. T., and Fu, H.
(1999) Biochemistry 38, 12159 –12164
20. Henriksson, M. L., Troller, U., and Hallberg, B. (2000) Biochem. J. 349,
697–701
21. Ichijo, H., Nishida, E., Irie, K., ten Dijke, P., Saitoh, M., Moriguchi, T., Takagi,
M., Matsumoto, K., Miyazono, K., and Gotoh, Y. (1997) Science 275, 90 –94
22. Marte, B. M., and Downward, J. (1997) Trends Biochem. Sci. 22, 355–358
23. del Peso, L., Gonzalez-Garcia, M., Page, C., Herrera, R., and Nunez, G. (1997)
Science 278, 687– 689
24. Datta, S. R., Dudek, H., Tao, X., Masters, S., Fu, H., Gotoh, Y., and Greenberg,
M. E. (1997) Cell 91, 231–241
25. Fisher, D. E. (1994) Cell 78, 539 –542
26. Kaufmann, S. H., and Gores, G. J. (2000) Bioessays 22, 1007–1017
27. Lowe, S. W., Ruley, H. E., Jacks, T., and Housman, D. E. (1993) Cell 74,
957–967
28. van Heusden, G. P., Griffiths, D. J., Ford, J. C., Chin, A. W. T. F., Schrader,
P. A., Carr, A. M., and Steensma, H. Y. (1995) Eur. J. Biochem. 229, 45–53
29. Ford, J. C., al-Khodairy, F., Fotou, E., Sheldrick, K. S., Griffiths, D. J., and
Carr, A. M. (1994) Science 265, 533–535
30. Gelperin, D., Weigle, J., Nelson, K., Roseboom, P., Irie, K., Matsumoto, K., and
Lemmon, S. (1995) Proc. Natl. Acad. Sci. U. S. A. 92, 11539 –11543
31. Harrington, E. A., Bennett, M. R., Fanidi, A., and Evan, G. I. (1994) EMBO J.
13, 3286 –3295
32. Masters, S. C., Yang, H., Datta, S. R., Greenberg, M. E., and Fu, H. (2001) Mol.
Pharmacol., in press

